index.js

```diff
@@ -206,7 +206,12 @@ exports.check_fcrdns = function (connection, results, next) {
         }
 
         // FCrDNS? PTR -> (A | AAAA) 3. PTR comparison
-        plugin.ptr_compare(results[fdom], connection, fdom)
+        if (Array.isArray(results[fdom])) {
+            plugin.ptr_compare(results[fdom], connection, fdom)
+        }
+        else {
+            plugin.ptr_compare([results[fdom]], connection, fdom)
```

Comment:
I've not seen a string get passed here but I am quite interested in seeing such a case.  If a string is getting found, a better fix is almost certainly to be found further upstream of this.

Reply:
And if you can help me replicate this, I'll write the test case.

Reply:
Two of the current tests cause a string to be passed (foo.example.com). Add the following console.log in ptr_compare(), run the test, and let the fun begin.

    for (var j=0; j<ip_list.length; j++) {
    	**console.log('ip_list[' + j + '] - ' + ip_list[j])**
        connection.results.push(plugin, {other_ips: ip_list[j]})
    }

I agree that upstream would be the better place to fix it but since I'm a bit time constrained I figured a quick fix would be better than no fix.

Reply:
Grrr, so much for my attempt at using the markdown feature. Please ignore the asterisks.

Reply:
Ahh, it's the test that is passing in an invalid result object. If you'll revert the Array test here, I'll merge this PR and fix the test. Alternatively, you can apply this patch which fixes the test and adds a few more:

```diff
index ca78762..1b03f7f 100644
--- a/test/index.js
+++ b/test/index.js
@@ -189,7 +189,8 @@ describe('check_fcrdns', function () {
 
     it('fail, tolerate', function (done) {
         this.connection.remote.ip = '10.1.1.1'
-        this.plugin.check_fcrdns(this.connection, ['foo.example.com'], function (rc, msg) {
+        const results = { 'foo.example.com': [ ] };
+        this.plugin.check_fcrdns(this.connection, results, function (rc, msg) {
             assert.equal(rc, undefined)
             done()
         })
@@ -198,7 +199,8 @@ describe('check_fcrdns', function () {
     it('null host', function (done) {
         // this result was experienced "in the wild"
         this.connection.remote.ip = '10.1.1.1'
-        this.plugin.check_fcrdns(this.connection, ['foo.example.com','', null], function (rc, msg) {
+        const results = { 'foo.example.com': [ '', null ] }
+        this.plugin.check_fcrdns(this.connection, results, function (rc, msg) {
             assert.equal(rc, undefined)
             done()
         })
@@ -207,17 +209,30 @@ describe('check_fcrdns', function () {
 
 describe('do_dns_lookups', function () {
 
-    it('performs a rdns lookup', function (done) {
-
-        const conn = this.connection
-        conn.remote.ip = '8.8.4.4'
-
-        this.plugin.do_dns_lookups((rc, msg) => {
-            assert.ok( /google.com/.test(conn.results.get('fcrdns').fcrdns[0]))
-            assert.equal(rc, undefined)
-            assert.equal(msg, undefined)
-            done()
-        },
-        conn)
+    const testIps = {
+        '8.8.4.4': 'google.com',
+        '2001:4860:4860::8844': 'google.com',
+        '4.2.2.2': 'level3.net',
+        '208.67.222.222': 'opendns.com',
+        // '2001:428::1': 'qwest.net',
+    }
+
+    Object.keys(testIps).forEach((ip) => {
+
+        it(`looks up ${ip}`, function (done) {
+
+            const conn = this.connection
+            conn.remote.ip = ip
+
+            this.plugin.do_dns_lookups((rc, msg) => {
+                const res = conn.results.get('fcrdns')
+                assert.ok(new RegExp( testIps[ip] ).test(res.fcrdns[0]))
+                // console.log(res);
+                assert.equal(rc, undefined)
+                assert.equal(msg, undefined)
+                done()
+            },
+            conn)
+        })
     })
 })
```